store/src/main/java/com/zextras/mailbox/tracking/PostHogTracking.java

```diff
@@ -18,7 +19,10 @@ public PostHogTracking(String endPoint) {
 
     @Override
     public void sendEventIgnoringFailure(Event event) {
-        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+        var clientBuilder = HttpClientBuilder.create();
+        HttpProxyUtil.configureProxy(clientBuilder);
+
+        try (CloseableHttpClient client = clientBuilder.build()) {
```

Comment:
I suggest using a factory to avoid such strong cohesion, for example by making PostHogTracking depend on a httpClient factory. In this case the real factory could return an httpclient already configured with all our needs.
```suggestion
        try (CloseableHttpClient client = httpClientFactory.create()) {
```

Reply:
We will try this approach. Thanks